beets/util/pipeline.py

```diff
@@ -415,8 +446,18 @@ def run_parallel(self, queue_size=DEFAULT_QUEUE_SIZE):
             # Make completely sure that all the threads have finished
             # before we return. They should already be either finished,
             # in normal operation, or aborted, in case of an exception.
-            for thread in threads[:-1]:
-                thread.join()
+            try:
+                for thread in threads[:-1]:
```

Comment:
Hmm, maybe so. Perhaps this is a case where "try it and see" would be a good idea?